src/main/java/io/github/vcuswimlab/stackintheflow/model/L2H/L2HPredictor.java

```diff
@@ -0,0 +1,209 @@
+		}
+	}
+
+	private static double[][] computeInitPredictions(int[][] newWords, int numUniqTags) {
+        double[][] initPredictions = new double[newWords.length][];
+        double p = 1./ (double) numUniqTags;
+        for (int i=0; i<newWords.length; i++) {
+            initPredictions[i] = new double[numUniqTags];
+            for (int j=0; j<numUniqTags; j++) {
+                initPredictions[i][j] = p;
+            }
+        }
+        return initPredictions;
+    }
+
+	private Map<String, Integer> nameToIdMap;
+	private CustomL2H testSampler;
+	private LabelTextDataset data;
+
+	public L2HPredictor() throws IOException {
+		nameToIdMap = new HashMap<>();
+		BufferedReader br = new BufferedReader(new InputStreamReader(this.getClass().getClassLoader().getResourceAsStream(Paths.get(formatFolder, formatFile+".wvoc").toString())));
+		int maxId = -1;
+		try {
+			String line;
+			int intId = 0;
+			while ((line = br.readLine()) != null) {
+				nameToIdMap.put(line, intId);
+				intId++;
+			}
+		} finally {
+			br.close();
+		}
+
+		data = new LabelTextDataset(datasetName);
+        data.setFormatFilename(formatFile);
+        data.loadFormattedData(formatFolder);
+        data.filterLabelsByFrequency(minLabelFreq);
+		data.prepareTopicCoherence(numTopWords);
+
+		testSampler = new CustomL2H();
+		testSampler.setVerbose(true);
+        testSampler.setDebug(false);
+        testSampler.setLog(false);
+        testSampler.setReport(false);
+        testSampler.configure("./output", data.getWordVocab().size(), alpha, beta, a0, b0,
+				new MSTBuilder(data.getLabels(), data.getLabelVocab()), false, false, AbstractSampler.InitialState.PRESET, PathAssumption.MAXIMAL, false, burnIn, modelIterations, sampleLag, 1);
+        testSampler.setTestConfigurations(burnIn, modelIterations, sampleLag);
+	}
+
+	public String computeL2HPredictions(String fileToPredict) throws Exception {
+		BufferedReader br;
+		br = new BufferedReader(new InputStreamReader(this.getClass().getClassLoader().getResourceAsStream(fileToPredict)));
+		Map<Integer, Integer> wordOccurrenceMap = new HashMap<>();
+		try {
+			String line;
+			while ((line = br.readLine()) != null) {
+				String[] parts = line.split("\\s+");
+				for(int i = 1; i < parts.length; i++) {
+					String name = parts[i-1] + " " + parts[i];
+					if(nameToIdMap.containsKey(name)) {
+						int index = nameToIdMap.get(name);
+						if(wordOccurrenceMap.containsKey(index)) {
+							wordOccurrenceMap.put(index, wordOccurrenceMap.get(index)+1);
+						} else {
+							wordOccurrenceMap.put(index, 1);
+						}
+					}
+				}
+			}
+		} finally {
+			br.close();
+		}
+
+		StringBuilder builder = new StringBuilder();
+		builder.append(wordOccurrenceMap.size() + " ");
+		for(Map.Entry<Integer,Integer> e : wordOccurrenceMap.entrySet()) {
+			builder.append(e.getKey() + ":" + e.getValue() + " ");
+		}
+
+		int[][] newWords = new TextDataLoader(new ByteArrayInputStream(builder.toString().getBytes())).getWords();
```

Comment:
I'm confused as to why you are packing a map into a string just so you can then unpack it into a 2d array, there are much more efficient ways of doing this, for example https://stackoverflow.com/questions/2265266/convert-hash-map-to-2d-array